src/Compilers/Core/Portable/Emit/CommonPEModuleBuilder.cs

```diff
@@ -1072,7 +1083,7 @@ public ArrayMethods ArrayMethods
 
 #nullable enable
 
-        internal PrivateImplementationDetails GetPrivateImplClass(TSyntaxNode syntaxNodeOpt, DiagnosticBag diagnostics)
+        internal PrivateImplementationDetails GetPrivateImplClass(TSyntaxNode? syntaxNode, DiagnosticBag diagnostics)
```

Comment:
>syntaxNode [](http://example.com/codeflow?start=79&length=10)

It doesn't look like the rename is necessary. In fact, I think the original name is more descriptive

Reply:
By adding the nullable annotation, `syntaxNodeOpt` triggers `RS0046: Avoid the 'Opt' suffix in nullable-enable code` - it's not a build warning, just a hint, but looks like our own rule we want to follow?

Reply:
FWIW, this rule is part of the Roslyn.Diagnostics.Analyzers (it has the RS prefix in the id) which are analyzers specific to dotnet/roslyn repository per [the docs](https://github.com/dotnet/roslyn-analyzers?tab=readme-ov-file#roslyndiagnosticsanalyzers). That's what made me think it's our own rule. But either way works for me, will revert the name change.